wall_e/extensions/ban.py

```diff
@@ -361,7 +364,14 @@ async def ban(self, ctx, user: discord.Member, *args):
         )
 
         # update database
-        await BanRecord.insert_record(ban)
+        success = await BanRecord.insert_record(ban)
+        if not success:
+            self.logger.info(f"[Ban ban()] Ban for {user} is duplicate.")
+            await self.mod_channel.send(f"Ban for {user} is duplicate")
+
+        # begin purging messages from user in the last purge_window_days
```

Comment:
honestly, I am a big fan of most of the logic in the ban command being moved till after a confirmation has been received in the db that the ban was successful.